java/code/src/com/redhat/rhn/frontend/strings/jsp/StringResource_en_US.xml

```diff
@@ -24890,6 +24890,9 @@ given channel.</source>
       <trans-unit id="overview.jsp.transactionalupdate.reboot" xml:space="preserve">
         <source>There is a pending transaction for this system, please reboot it to activate the changes.</source>
       </trans-unit>
+      <trans-unit id="overview.jsp.transactionalupdate.reboot-static" xml:space="preserve">
+        <source>OS with transactional updates: please reboot after any packages and/or channels changes.</source>
```

Comment:
Channel change? Isn't channel changing just repositories in `/etc/zypp`? That should be fine as it is in overlay and is not creating new snapshot.

Reply:
This is good question. Technically correct is to suggest reboot when new snapshot is created as without reboot it can lead to data loss.
New snapshot is created by running `transactional-update` or `tukit`. But salt transactional-update executor covers many states and runs it always through `transactional-update` which may or may not create snapshot depending if there was a change in watched subvolumes (e.g. someone uses `file.managed` on file in `/usr/`). We can't tell when new transaction is created if beacon is not running yet.

So either we go with `reboot after any package change` or be paranoid and go with `reboot after any salt state application` or `reboot after doing any change on the system`.

Reply:
Does changing the channels generate a snapshot? I'd expect that's not the case as the changes should happen at /etc/zypp?

About the other alternatives, I don't have a strong opinion. Either we cover too little, wither we cover too much (even with `after any salt state application`).

Reply:
I get the point Ondrej is making. I would suggest the following to update the message

`OS with transactional updates:Please reboot after any change in order to apply them.`

I would also suggest to not merge this PR till tomorrow at least and see if we could get the fix for the beacon issue.

Reply:
> Does changing the channels generate a snapshot? I'd expect that's not the case as the changes should happen at /etc/zypp?

It should not keep it. When we add channels, we are doing `state.apply channels`. `state.apply` is one of the delegated functions to `transactional_update.apply` ([see](https://github.com/saltstack/salt/blob/master/salt/executors/transactional_update.py#L17)) and `transactional_update.apply` calls `salt-call` through `transactional-update run -d`. `-d` means it will drop the snapshot if no change in snapshot is detected.

So it creates new snapshot, but because modifying repositories modifies only `/etc` in overlay it is not detected as change in snapshot and changes are dropped, however changes in overlay are moved back to real `/etc` (here is the bug in transactional_update and potential data loss if you do not reboot after snapshot is created. It should move data to original snapshot, not currently running one and all gets complicated if users ignore reboot requests).